pyxrf/model/draw_image_rgb.py

```diff
@@ -0,0 +1,353 @@
+        self.fig = plt.figure(figsize=(6,6))
+        self.rgb_name_list = ['R', 'G', 'B']
+
+    def data_dict_update(self, change):
+        """
+        Observer function to be connected to the fileio model
+        in the top-level gui.py startup
+
+        Parameters
+        ----------
+        changed : dict
+            This is the dictionary that gets passed to a function
+            with the @observe decorator
+        """
+        self.data_dict = change['value']
+
+    @observe('data_dict', 'data_dict_keys')
+    def init_plot_status(self, change):
+        # initiate the plotting status once new data is coming
+        self.data_opt = 0
+        self.rgb_name_list = ['R', 'G', 'B']
+        self.index_red = 0
+        self.index_green = 1
+        self.index_blue = 2
+
+        # init of scaler for normalization
+        self.scaler_name_index = 0
+
+        self.data_dict_keys = self.data_dict.keys()
+        logger.info('The following groups are included for 2D image display: {}'.format(self.data_dict_keys))
+
+        scaler_groups = [v for v in self.data_dict.keys() if 'scaler' in v]
+        if len(scaler_groups) > 0:
+            #self.scaler_group_name = scaler_groups[0]
+            self.scaler_norm_dict = self.data_dict[scaler_groups[0]]
+            # for GUI purpose only
+            self.scaler_items = []
+            self.scaler_items = self.scaler_norm_dict.keys()
+            self.scaler_data = None
+
+        self.show_image()
+
+    @observe('data_opt')
+    def _update_file(self, change):
+        if self.data_opt == 0:
+            self.dict_to_plot = {}
+            self.items_in_selected_group = self.dict_to_plot.keys()
+            self.set_stat_for_all(bool_val=False)
+        elif self.data_opt > 0:
+            self.set_stat_for_all(bool_val=False)
+            plot_item = sorted(self.data_dict_keys)[self.data_opt-1]
+            self.dict_to_plot = self.data_dict[plot_item]
+            # for GUI purpose only
+            self.items_in_selected_group = []
+            self.items_in_selected_group = self.dict_to_plot.keys()
+            self.set_stat_for_all(bool_val=False)
+
+    @observe('scaler_name_index')
+    def _get_scaler_data(self, change):
+        if self.scaler_name_index == 0:
+            self.scaler_data = None
+        else:
+            scaler_name = self.scaler_items[self.scaler_name_index-1]
+            #self.scaler_data = self.data_dict[self.scaler_group_name][scaler_name]
+            self.scaler_data = self.scaler_norm_dict[scaler_name]
+            logger.info('Use scaler data to normalize,'
+                        'and the shape of scaler data is {}'.format(self.scaler_data.shape))
+        self.show_image()
+
+    @observe('scale_opt')
+    def _update_scale(self, change):
+        if change['type'] != 'create':
+            self.show_image()
+
+    def set_stat_for_all(self, bool_val=False):
+        """
+        Set plotting status for all the 2D images.
```

Comment:
It is related to the gui level, i.e., if the checkbox is clicked, the plotting status is true.